### ISSUE TITLE
Drag of clip on timeline with bad position

### ISSUE THREAD

**Issue by mikecrash** [mltframework/shotcut#1380]
When dragging clip on time line, the current behavior is this:
1. press left mouse button on the clip
2. move mouse to the left/right to move the position, nothing happens until you move the mouse by some amount (about 20 pixels?)
3. clip starts moving not from the position where button pressed, but on the position of current mouse position when dragging starts, if you move mouse fast, you drag clip outside of the clip, so clip is not on the position of mouse, but somewhere far away from cursor

Correct behavior:
Clip holds on mouse position when first clicked

Hint:
It seems when the dragging starts, the mouse position is saved and from that position is counted the moved position of the clip. However correct function should be to save the start mouse position in onmousedown event and from that position count the new position of clip.

Tested version:
22.11.25

**Comment by bmatherly** [comment]
This does not match my experience when snapping is turned off. Maybe you have snapping on?

![image](https://user-images.githubusercontent.com/821968/205536161-cf86c773-5022-41af-8fa3-09a8e6b6246e.png)

**Comment by mikecrash** [comment]
It doesn't matter if the snap is on or off. Most significant it is when moving the mouse fast.

May be it has something to do with OS? Tested on Debian in GNOME. I will test it in Windows.

**Comment by luzpaz** [comment]
@mikecrash can you provide a screencast of this occuring perhaps ?

**Comment by ddennedy** [comment]
@luzpaz stop bumping our bugs with your comments when you are never the one to fix one or provide more insight.

**Comment by luzpaz** [comment]
@mikecrash can you still reproduce on 23.09 ?

I can't reproduce on 
Manjaro Linux 23.02
Shotcut 23.09

**Comment by mikecrash** [comment]
@luzpaz Yes browser says corrupt, but if I download the video and play it with vlc, it works. Try the same...

**Comment by luzpaz** [comment]
Downloaded. Shows black screen for 8 seconds.

**Comment by mikecrash** [comment]
Opened in ShotCut and reexeported. Hope it will work.

https://github.com/mltframework/shotcut/assets/60000811/831938d9-126b-493e-bb73-cd47e6d75251
